packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -614,25 +712,84 @@ class WidgetInspectorService {
     return _followDiagnosticableChain(chain.reversed.toList());
   }
 
-  Map<String, Object> _nodeToJson(DiagnosticsNode node, String groupName) {
+  Map<String, Object> _nodeToJson(
+    DiagnosticsNode node,
+    _SerializeConfig config,
+  ) {
     if (node == null)
       return null;
     final Map<String, Object> json = node.toJsonMap();
 
-    json['objectId'] = toId(node, groupName);
+    json['objectId'] = toId(node, config.groupName);
     final Object value = node.value;
-    json['valueId'] = toId(value, groupName);
+    json['valueId'] = toId(value, config.groupName);
+
+    if (config.summaryTree) {
+      json['summaryTree'] = true;
+    }
 
     final _Location creationLocation = _getCreationLocation(value);
+    bool createdByLocalProject = false;
     if (creationLocation != null) {
       json['creationLocation'] = creationLocation.toJsonMap();
       if (_isLocalCreationLocation(creationLocation)) {
+        createdByLocalProject = true;
         json['createdByLocalProject'] = true;
       }
     }
+
+    if (config.subtreeDepth > 0 ||
+        (config.pathToInclude != null && config.pathToInclude.isNotEmpty)) {
+      json['children'] = _nodesToJson(_getChildrenHelper(node, config), config);
+    }
+
+    if (config.includeProperties) {
+      json['properties'] = _nodesToJson(
+        node.getProperties().where(
+          (DiagnosticsNode node) => !node.isFiltered(createdByLocalProject ? DiagnosticLevel.fine : DiagnosticLevel.info),
+        ),
+        new _SerializeConfig(groupName: config.groupName, subtreeDepth: 1, expandPropertyValues: true),
+      );
+    }
+
+    if (node is DiagnosticsProperty) {
+      // Add additional information about properties needed for graphical
+      // display of properties.
+      if (value is Color) {
+        json['valueProperties'] = <String, Object>{
+          'red': value.red,
+          'green': value.green,
+          'blue': value.blue,
+          'alpha': value.alpha,
+        };
+      } else if (value is IconData) {
+        json['valueProperties'] = <String, Object>{
+          'codePoint': value.codePoint,
+        };
+      }
```

Comment:
I'm not a big fan of special-casing specific types here, but I don't have a better solution for you.